### PR TITLE
m4ri: init at 20140914

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Package m4ri.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/science/math/m4ri/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = http://m4ri.sagemath.org/;
+    description = "Library for matrix multiplication, reduction and inversion over GF(2)";
+    longDescription = ''
+      M4RIE is a library for fast arithmetic with dense matrices over small finite fields of even characteristic. It uses the M4RI library, implementing the same operations over the finite field F2.
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/m4ri/default.nix]: That's the description for m4rie.

File: pkgs/development/libraries/science/math/m4ri/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    '';
+    license = with licenses; [ gpl2 ];
+    maintainers = with maintainers; [ timokau ];
+    platforms = platforms.all;
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/m4ri/default.nix]: Have you tested all platforms? It's generally better to set to Linux only or to `platforms.unix` if Darwin was also tested.

File: pkgs/development/libraries/science/math/m4ri/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  doCheck = true;
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = http://m4ri.sagemath.org/;
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/m4ri/default.nix]: seems to be down

**Comment by dotlambda** [comment]
@GrahamcOfBorg build m4ri

File: pkgs/development/libraries/science/math/m4ri/default.nix (hunk 2 above)
Comment (timokau) [reply pkgs/development/libraries/science/math/m4ri/default.nix]: No, but I know that it works on various platforms in the sage distribution (including ARM, darwin, cygwin) so I have reasonable hope that it'll work.

Comment (dotlambda) [reply pkgs/development/libraries/science/math/m4ri/default.nix]: Please set `platforms = platforms.linux` until someone builds this on macOS.

File: pkgs/development/libraries/science/math/m4ri/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [
+    autoreconfHook
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://malb.bitbucket.io/m4ri/;
+    description = "Library to do fast arithmetic with dense matrices over F_2";
+    longDescription = ''
+      M4RIE is a library for fast arithmetic with dense matrices over small finite fields of even characteristic. It uses the M4RI library, implementing the same operations over the finite field F2.
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/m4ri/default.nix]: That's still the wrong `longDescription`.

Comment (timokau) [reply pkgs/development/libraries/science/math/m4ri/default.nix]: Oh, I meant to delete that.

**Comment by timokau** [comment]
Thanks for taking the time to review and sorry for my oversights @dotlambda.

**Pull request merged**